Test failures on node v14-v16

I get the following test failure on Linux after fixing the macOS failures in #393 with PR #394 

```
  1) socket cleanup
       should cleanup sockets on ERR_STREAM_PREMATURE_CLOSE (using stream.pipeline):

      Uncaught AssertionError: expected 1 to equal 0
      + expected - actual

      -1
      +0
      
      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value (node_modules/should/cjs/should.js:356:19)
      at Timeout._onTimeout (test/socket_cleanup_spec.js:73:33)
      at listOnTimeout (node:internal/timers:557:17)
      at processTimers (node:internal/timers:500:7)
```

--- a/test/socket_cleanup_spec.js
+++ b/test/socket_cleanup_spec.js
@@ -61,7 +61,7 @@ describe('socket cleanup', function(){
 
     stream.pipeline(resp, writable, function(err) {
       err.code.should.eql('ERR_STREAM_PREMATURE_CLOSE')
-      // if (err) resp.request.destroy();
+      if (err) resp.request.destroy();
     });
 
     setTimeout(function() {

